source/time_stepping/convection_time_step.cc

```diff
@@ -33,16 +33,39 @@ namespace aspect
       const QIterated<dim> quadrature_formula (QTrapezoid<1>(),
                                                this->get_parameters().stokes_velocity_degree);
 
+      bool consider_darcy_timestep = false;
+      if (this->introspection().composition_type_exists(CompositionalFieldDescription::porosity))
+        {
+          consider_darcy_timestep = this->get_parameters().compositional_field_methods[this->introspection().compositional_index_for_name("porosity")] == Parameters<dim>::AdvectionFieldMethod::fem_darcy_field
+                                    ?
+                                    true
+                                    :
+                                    false;
+        }
```

Comment:
What about this:
```suggestion
      bool consider_darcy_timestep = false;
      unsigned int porosity_idx = numbers::invalid_unsigned_int;
      if (this->introspection().composition_type_exists(CompositionalFieldDescription::porosity))
      {
         porosity_idx = this->introspection().find_composition_type(CompositionalFieldDescription::porosity);
        if (this->get_parameters().compositional_field_methods[porosity_idx] == Parameters<dim>::AdvectionFieldMethod::fem_darcy_field)
          consider_darcy_timestep = true;
      }
```

Reply:
Ahhhh this makes sense, I was hesitant to make porosity_idx not const